ow_plexil/src/plexil-adapter/OwlatInterface.cpp

```diff
@@ -109,3 +294,562 @@ void OwlatInterface::owlatStowAction (int id)
+  goal.frame.value = frame;
+  goal.relative = relative; 
+  for(int i = 0; i < goal.point.size(); i++){
+    goal.point[i] = point[i]; 
+    goal.normal[i] = normal[i];
+  }  
+  goal.preload = preload;
+  goal.max_torque = max_torque;
+  string opname = Name_OwlatTaskShearBevameter;  // shorter version
+
+  runAction<actionlib::SimpleActionClient<TASK_SHEAR_BEVAMETERAction>,
+            TASK_SHEAR_BEVAMETERGoal,
+            TASK_SHEAR_BEVAMETERResultConstPtr,
+            TASK_SHEAR_BEVAMETERFeedbackConstPtr>
+    (opname, m_owlatTaskShearBevameterClient, goal, id,
+     default_action_active_cb (opname),
+     default_action_feedback_cb<TASK_SHEAR_BEVAMETERFeedbackConstPtr> (opname),
+     task_shear_bevameter_done_cb<OwlatArmTaskShearBevameter, TASK_SHEAR_BEVAMETERResultConstPtr>);
+}
+
+
+void OwlatInterface::armJointAnglesCallback(const owlat_sim_msgs::ARM_JOINT_ANGLES::ConstPtr& msg)
+{
+  std::copy(msg->value.begin(), msg->value.end(), m_arm_joint_angles.begin()); 
+  publish("ArmJointAngles", m_arm_joint_angles);
+}
+
+void OwlatInterface::armJointAccelerationsCallback(const owlat_sim_msgs::ARM_JOINT_ACCELERATIONS::ConstPtr& msg)
+{
+  std::copy(msg->value.begin(), msg->value.end(), m_arm_joint_accelerations.begin()); 
+  publish("ArmJointAccelerations", m_arm_joint_accelerations);
+}
+
+void OwlatInterface::armJointTorquesCallback(const owlat_sim_msgs::ARM_JOINT_TORQUES::ConstPtr& msg)
+{
+  std::copy(msg->value.begin(), msg->value.end(), m_arm_joint_torques.begin()); 
+  publish("ArmJointTorques", m_arm_joint_torques);
+}
+
+void OwlatInterface::armJointVelocitiesCallback(const owlat_sim_msgs::ARM_JOINT_VELOCITIES::ConstPtr& msg)
+{
+  std::copy(msg->value.begin(), msg->value.end(), m_arm_joint_velocities.begin()); 
+  publish("ArmJointVelocities", m_arm_joint_velocities);
+}
+
+void OwlatInterface::armFTTorqueCallback(const owlat_sim_msgs::ARM_FT_TORQUE::ConstPtr& msg)
+{
+  std::copy(msg->value.begin(), msg->value.end(), m_arm_ft_torque.begin()); 
+  publish("ArmFTTorque", m_arm_ft_torque);
+}
+
+void OwlatInterface::armFTForceCallback(const owlat_sim_msgs::ARM_FT_FORCE::ConstPtr& msg)
+{
+  std::copy(msg->value.begin(), msg->value.end(), m_arm_ft_force.begin()); 
+  publish("ArmFTForce", m_arm_ft_force);
+}
+
+void OwlatInterface::armPoseCallback(const owlat_sim_msgs::ARM_POSE::ConstPtr& msg)
+{
+  m_arm_pose[0] = msg->value.position.x;
+  m_arm_pose[1] = msg->value.position.y;
+  m_arm_pose[2] = msg->value.position.z;
+  m_arm_pose[3] = msg->value.orientation.x;
+  m_arm_pose[4] = msg->value.orientation.y;
+  m_arm_pose[5] = msg->value.orientation.z;
+  m_arm_pose[6] = msg->value.orientation.w;
+  publish("ArmPose", m_arm_pose);
+}
+
+void OwlatInterface::armToolCallback(const owlat_sim_msgs::ARM_TOOL::ConstPtr& msg)
+{
+  m_arm_tool = msg->value.value;
+  publish("ArmTool", m_arm_tool);
+}
+
+Value OwlatInterface::getArmJointAngles()
+{
+  Value value_joint_angles = Value(m_arm_joint_angles);
+  return value_joint_angles;
```

Comment:
In this and all the following functions like it, it is shorter and possibly more efficient (via _return value optimization_) to just do `return Value(m_arm_joint_angles);`